### PR TITLE
test: unit test action runner

### PR DESCRIPTION
**Pull request by realSpok**
## Definition of Done
#86bx3056z

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/cloud-toolkit/test/services/action-runner.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        action.mockRejectedValue(new BusinessError(message));
+
+        await actionRunner(action)(args);
+
+        expect(spinner.fail).toHaveBeenCalled();
+        expect(spinner.fail).toHaveBeenCalledWith(message);
+      });
+    });
+
+    describe('when an uncaught Error is thrown', () => {
```
Comment (Scra3) [inline packages/cloud-toolkit/test/services/action-runner.unit.test.ts:50]: => when a non Business error is thrown

File: packages/cloud-toolkit/test/services/action-runner.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      it('should let it throw', async () => {
+        const { action, args, spinner } = setup();
+        action.mockRejectedValue(new Error('some error occured'));
+
+        await expect(actionRunner(action)(args)).rejects.toEqual(new Error('some error occured'));
+
+        expect(spinner.fail).not.toHaveBeenCalled();
```
Comment (Scra3) [inline packages/cloud-toolkit/test/services/action-runner.unit.test.ts:57]: We should check if "stop" is called.

File: packages/cloud-toolkit/test/services/action-runner.unit.test.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    describe('when BusinessError is thrown', () => {
+      it('should call fail on the spinner', async () => {
+        const { action, args, spinner } = setup();
+        const message = 'some business error occured';
+        action.mockRejectedValue(new BusinessError(message));
+
+        await actionRunner(action)(args);
+
+        expect(spinner.fail).toHaveBeenCalled();
+        expect(spinner.fail).toHaveBeenCalledWith(message);
```
Comment (Scra3) [inline packages/cloud-toolkit/test/services/action-runner.unit.test.ts:46]: We should check if "stop" is called.

File: packages/cloud-toolkit/test/services/action-runner.unit.test.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  describe('when the action succeeds', () => {
+    it('should return a function starting the spinner', async () => {
+      const { action, args, spinner } = setup();
+
+      await actionRunner(action)(args);
+
+      expect(action).toHaveBeenCalled();
+      expect(action).toHaveBeenCalledWith(spinner, args);
+      expect(spinner.fail).not.toHaveBeenCalled();
```
Comment (Scra3) [inline packages/cloud-toolkit/test/services/action-runner.unit.test.ts:32]: We should check if "stop" is called and "start"

Comment (realSpok) [reply packages/cloud-toolkit/test/services/action-runner.unit.test.ts:32]: yes for stop.
start is what returns the spinner, so it needs to have been called. I'm not sure we need to test that

**Review by Scra3 — APPROVED** [review]

**Pull request merged**